### PR TITLE
feat: Multimint support

### PR DESCRIPTION
**Pull request by Kodylow**
uses new multimint implementation. Going to put some helper functions into multimint first to simplify accessing a ClientArc by federation id as str but generally the approach. @benthecarman lmk if you like this. Will want to refactor to not have to drill down so much app state on the callback but will do that in a separate PR

### CODE REVIEW

**Comment by Kodylow** [comment]
uses new multimint implementation. Going to put some helper functions into multimint first to simplify accessing a ClientArc by federation id as str but generally the approach. @benthecarman lmk if you like this. Will want to refactor to not have to drill down so much app state on the callback but will do that in a separate PR

**Comment by benthecarman** [comment]
LFG will review

File: src/router/handlers/nostr/register.rs [hunk 1]
```diff
@@ -17,6 +18,7 @@ pub struct UserParams {
     pub pubkey: String,
     pub name: String,
     pub dm_type: SupportedDmType,
+    pub federation_id: FederationId,
```
Comment (benthecarman) [inline src/router/handlers/nostr/register.rs:21]: what if the federation isn't configured on the lightning address server? would it be possible to add the federation on the fly

File: src/router/handlers/lnurlp/callback.rs [hunk 2]
```diff
@@ -105,8 +106,23 @@ pub async fn handle_callback(
...
+            StatusCode::BAD_REQUEST,
+            anyhow::anyhow!("Invalid federation_id: {}", e),
+        )
+    })?;
+
+    let locked_clients = state.fm.clients.lock().await.clone();
+    let client = locked_clients.get(&federation_id).ok_or_else(|| {
+        AppError::new(
+            StatusCode::BAD_REQUEST,
+            anyhow::anyhow!("FederationId not found in multimint map"),
```
Comment (benthecarman) [inline src/router/handlers/lnurlp/callback.rs:120]: i feel like it'd be better to have a db of invite codes so we can just connect to it when needed

**Review by benthecarman — COMMENTED** [review]
otherwise lgtm, would be great if you could add federations on the fly so the server doesn't need to be updated for every new federation

**Comment by Kodylow** [comment]
When you say adding federations on the fly at what point would that happen? I was thinking the registration step. could have it that the user provides an invite code for their federation as part of that, connect if not already a client.

Multimint creates a different db for each clientArc (each fed client) and reloads them on restart.

**Pull request merged**